tcl/html2tcl.tcl

```diff
@@ -0,0 +1,85 @@
+namespace eval qc {
+    package require tdom
+    namespace export h h_tag tcl_escape tdom_node2tcl html2tcl
+}
+
+proc qc::h_tag { tag_name args } {
+    #| Generate just the opening html tag
+    set singleton_tags [list area base br col command embed hr img input link meta param source]
+    set minimized [list compact checked declare readonly disabled selected defer ismap nohref noshade nowrap multiple noresize]
+    set attributes {}
+   
+    foreach {name value} [qc::dict_exclude $args {*}$minimized] {
+	lappend attributes "$name=\"[string map {< &lt; > &gt; & &amp; \" &#34;} $value]\""
+    }
+    foreach {name value} [qc::dict_subset $args {*}$minimized] {
+	if { [string is true $value] } {
+	    lappend attributes "$name=\"$name\""
+	}
+    }
+
+    if { $tag_name in $singleton_tags && [llength $attributes]==0 } { 
+	return "<$tag_name/>"
+    } elseif { $tag_name in $singleton_tags } { 
+        return "<$tag_name [join $attributes]/>"
+    } elseif { [llength $attributes]==0 } {
+	return "<$tag_name>"
+    } else {
+	return "<$tag_name [join $attributes]>"
+    }
+}
+
+proc qc::h {tag_name args} {
+    #| Generate an html node
```

Comment:
yes and I think we will drop qc::html and qc::html_tag from qcode-tcl-6 or at least deprecate.